### PR TITLE
Refactored API

### PR DESCRIPTION
**Pull request by nsvir**

### CODE REVIEW

File: src/main/java/io/github/anthorx/parquet/sql/api/JDBCReader.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import java.sql.*;
+
+public class JDBCReader implements AutoCloseable {
+
+  private final SQLField[] columns;
+  private final int columnCount;
+  private final PreparedStatement preparedStatement;
+  private final ResultSet resultSet;
+
+  public JDBCReader(DataSource dataSource, String tableName, int fetchSize) throws SQLException {
```
Comment (anthoRx) [inline src/main/java/io/github/anthorx/parquet/sql/api/JDBCReader.java:31]: As for the writer, Connection is enough here as you don't use the Datasource object

**Review by anthoRx — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 2]
```diff
@@ -133,13 +132,6 @@
             </exclusions>
             <scope>test</scope>
         </dependency>
-
```
[the remark this answers is not in the document]
Comment (nsvir) [reply pom.xml:136]: restored

File: src/main/java/io/github/anthorx/parquet/sql/parquet/model/Record.java [hunk 3]
```diff
@@ -46,6 +48,20 @@ public Record() {
         .findFirst();
   }
 
+  /**
+   * based on schemaField,
+   * extract recordField from record or create nullRecordField
+   *
+   * @return the full recordFields
+   */
+  public List<RecordField<?>> getRecordField(List<Type> schemaField) {
```
[the remark this answers is not in the document]
Comment (nsvir) [reply src/main/java/io/github/anthorx/parquet/sql/parquet/model/Record.java:57]: True, now method is named "readAll", takes a ReadRecordConsumer and apply it to each record fields, schemaField is still. used to find the field order

File: src/main/java/io/github/anthorx/parquet/sql/api/JDBCWriter.java [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+  private final PreparedStatement preparedStatement;
+  private final List<String> errors;
+  private int currentParameterIndex = 0;
+
+  /**
+   * default constructor that should be used
+   * If you want a custom prepare statement, use the other constructor
+   */
+  public JDBCWriter(DataSource dataSource, String tableName, Collection<String> columnNames) throws SQLException, IllegalArgumentException {
```
[the remark this answers is not in the document]
Comment (nsvir) [reply src/main/java/io/github/anthorx/parquet/sql/api/JDBCWriter.java:49]: true

File: src/main/java/io/github/anthorx/parquet/sql/api/JDBCReader.java (hunk 1 above)
Comment (nsvir) [reply src/main/java/io/github/anthorx/parquet/sql/api/JDBCReader.java:31]: true as well

**Review by anthoRx — APPROVED** [review]

**Pull request merged**